### ISSUE TITLE
Inserting array column into BigQuery

### ISSUE THREAD

**Issue by PabloPaglilla** [GoogleCloudDataproc/spark-bigquery-connector#101]
Hello!

I'm working on a PySpark job running on Dataproc that needs to insert a DataFrame into a BigQuery table. The DataFrame contains a column of type `ArrayType(StringType())`, which is calculated using a UDF the following way:

``` python
def parse_categories(event):
    # Business logic
    # Return a list of strings
    return ['a', 'list', 'of', 'strings']

parse_categories_udf = udf(parse_categories, ArrayType(StringType()))

df = load_data()
df = df.withColumn('categories', parse_categories_udf(df.event))
```

My table on BigQuery has the categories column as a `STRING` type with mode `REPEATED`. However, when I try to insert into the table this way:

``` python
df.write.format('bigquery') \
        .option('table', 'my_dataset.my_table') \
        .mode('append') \
        .save()
```

I get the following error:

```
java.lang.RuntimeException: Failed to write to BigQuery
    ....
Caused by: com.google.cloud.spark.bigquery.repackaged.com.google.cloud.bigquery.BigQueryException: Failed to load to my_dataset.my_table in job JobId{project=my-project, job=ff30135b-899e-4751-884e-d7fda57ce87e, location=US}.
BigQuery error was Provided Schema does not match Table my-project:my_dataset.my_table.
Field categories has changed type from STRING to RECORD
    ....
```

I understand this might be an issue with my implementation rather than with the connector, but I haven't found a way to get it to work.

Thanks in advance!

**Comment by PabloPaglilla** [comment]
Hello!

I've managed to solve the problem. It seems [load jobs from GCS to BigQuery don't support arrays when using Parquet format ](https://cloud.google.com/bigquery/docs/loading-data-cloud-storage-parquet#parquet_conversions), which the connector uses by default.

I set the `intermediateFormat` configuration so that the connector would use ORC format and it worked just fine. Sadly, the BigQuery error message was pretty misleading.

If I may ask, is there a reason to use parquet as the default format over ORC?
